src/main/java/com/example/auctionapp/dto/SearchRequest.java

```diff
@@ -0,0 +1,133 @@
+package com.example.auctionapp.dto;
+
+import com.example.auctionapp.customValidator.ColorConstraint;
+import com.example.auctionapp.customValidator.SizeConstraint;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+public class SearchRequest {
```

Comment:
try to think in advance, this search request will be reused in the listing of user products, etc if im not mistaken.
Maybe it can be reused if generalised.

Reply:
To be discussed later